list/list.js

```diff
@@ -1004,8 +1005,8 @@ assign(DefineList.prototype, {
 	map: function(callback, thisArg) {
 		var mappedList = [],
 			self = this;
-		this.each(function(item, index, list) {
-			var mapped = callback.call(thisArg | self, item, index, self);
+		canReflect.eachIndex(this, function(item, index, list) {
+			var mapped = canReflect.call(callback, thisArg | self, item, index, self);
```

Comment:
I guess the question on my mind was was how `apply`/`call` got added to can-reflect if we didn't really need them yet for non-natively-callable function-likes